services/requester/batch_tx_pool.go

```diff
@@ -0,0 +1,286 @@
+
+			for address, pooledTxs := range txsGroupedByAddress {
+				err := t.batchSubmitTransactionsForSameAddress(
+					latestBlock,
+					account,
+					pooledTxs,
+				)
+				if err != nil {
+					t.logger.Error().Err(err).Msgf(
+						"failed to send Flow transaction from BatchPool for EOA: %s",
+						address.Hex(),
+					)
+					continue
+				}
+			}
+		}
+	}
+}
+
+func (t *BatchTxPool) batchSubmitTransactionsForSameAddress(
+	latestBlock *flow.Block,
+	account *flow.Account,
+	pooledTxs []pooledEvmTx,
+) error {
+	// Sort the transactions based on their nonce, to make sure
+	// that no re-ordering has happened due to races etc.
+	sort.Slice(pooledTxs, func(i, j int) bool {
+		return pooledTxs[i].nonce < pooledTxs[j].nonce
+	})
+
+	hexEncodedTxs := make([]cadence.Value, len(pooledTxs))
+	for i, txPayload := range pooledTxs {
+		hexEncodedTxs[i] = txPayload.txPayload
+	}
+
+	coinbaseAddress, err := cadence.NewString(t.config.Coinbase.Hex())
+	if err != nil {
+		return err
+	}
+
+	script := replaceAddresses(batchRunTxScript, t.config.FlowNetworkID)
+	flowTx, err := t.buildTransaction(
+		latestBlock,
+		account,
+		script,
+		cadence.NewArray(hexEncodedTxs),
+		coinbaseAddress,
+	)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+	if err := t.client.SendTransaction(ctx, *flowTx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// buildTransaction creates a Cadence transaction from the provided script,
+// with the given arguments and signs it with the configured COA account.
+func (t *BatchTxPool) buildTransaction(
+	latestBlock *flow.Block,
+	account *flow.Account,
+	script []byte,
+	args ...cadence.Value,
+) (*flow.Transaction, error) {
+	defer func() {
+		t.collector.AvailableSigningKeys(t.keystore.AvailableKeys())
+	}()
+
+	flowTx := flow.NewTransaction().
+		SetScript(script).
+		SetReferenceBlockID(latestBlock.ID).
+		SetComputeLimit(flowGo.DefaultMaxTransactionGasLimit)
+
+	for _, arg := range args {
+		if err := flowTx.AddArgument(arg); err != nil {
+			return nil, fmt.Errorf("failed to add argument: %s, with %w", arg, err)
+		}
+	}
+
+	// building and signing transactions should be blocking,
+	// so we don't have keys conflict
+	t.mux.Lock()
+	defer t.mux.Unlock()
+
+	accKey, err := t.keystore.Take()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := accKey.SetProposerPayerAndSign(flowTx, account); err != nil {
+		accKey.Done()
+		return nil, err
+	}
+
+	// now that the transaction is prepared, store the transaction's metadata
+	accKey.SetLockMetadata(flowTx.ID(), latestBlock.Height)
+
+	t.collector.OperatorBalance(account)
+
+	return flowTx, nil
+}
+
+func (t *BatchTxPool) fetchFlowLatestBlockAndCOA() (
+	*flow.Block,
+	*flow.Account,
+	error,
+) {
+	var (
+		g           = errgroup.Group{}
+		err1, err2  error
+		latestBlock *flow.Block
+		account     *flow.Account
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
same here

Reply:
Good point :+1: Updated in https://github.com/onflow/flow-evm-gateway/pull/831/commits/2a98db83c0099a71511819ff0d75d3747fde5700 .
Also removed the `WithTimeout` addition, since `SingleTxPool` didn't use anything like that, and the timeout I added was a bit magic.